pymc/logprob/transforms.py

```diff
@@ -429,10 +441,16 @@ def find_measurable_transforms(fgraph: FunctionGraph, node: Node) -> Optional[Li
         return None
 
     [measurable_input] = measurable_inputs
+    [measurable_output] = node.outputs
 
-    # Do not apply rewrite to discrete variables
+    # Do not apply rewrite to discrete variables except for their addition and negation
     if measurable_input.type.dtype.startswith("int"):
-        return None
+        if not (
+            find_negated_var(measurable_output) is not None or isinstance(node.op.scalar_op, Add)
+        ):
+            return None
+        if not measurable_output.type.dtype.startswith("int"):
```

Comment:
```suggestion
        # Do not allow rewrite if output is cast to a float, because we don't have meta-info on the type of the MeasurableVariable
        if not measurable_output.type.dtype.startswith("int"):
```